### PR TITLE
Use graphiql option variables if query variables are not supplied

### LINKED ISSUE #193: Variables Passed to renderGraphiQL Are Occluded by Default Assignment
Using graphiqlExpress specifically here, but Hapi and Koa all have the same issue as well:
- https://github.com/apollostack/graphql-server/blob/master/packages/graphql-server-express/src/expressApollo.ts#L161
- https://github.com/apollostack/graphql-server/blob/master/packages/graphql-server-hapi/src/hapiApollo.ts#L227
- https://github.com/apollostack/graphql-server/blob/master/packages/graphql-server-koa/src/koaApollo.ts#L127

Given an invocation like this:

``` javascript
const initialQuery = `query exampleQuery ($name: String!) {
  person(name: $name) {
    id
    name
  }
}`;

const initialVariables = { name: 'Joe Bob' };

app.use('/graphiql', graphiqlExpress({
  endpointURL: '/graphql',
  query: initialQuery,
  variables: initialVariables,
}));
```

... your variables will never be used.

This is because an initial assignment is made inside of graphiqlExpress, etc, of:

``` javascript
const variables = q.variables || '{}';
```

And then renderGraphiQL is invoked with:

``` javascript
variables: JSON.parse(variables) || options.variables,
```

Because `variables` will always be an empty object if you didn't pass variables in the query string, `options.variables` will never be provided.

### PR DESCRIPTION
**Pull request by alanchristensen**
TODO:

- [x] Update CHANGELOG.md with your change (include reference to issue & this PR)
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

Closes #193

### CODE REVIEW

File: packages/graphql-server-hapi/src/hapiApollo.test.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import * as hapi from 'hapi';
-import { graphqlHapi, graphiqlHapi, HapiPluginOptions } from './hapiApollo';
+import { graphqlHapi, graphiqlHapi } from './hapiApollo';
 import 'mocha';
 
 import testSuite, { Schema } from 'graphql-server-integration-testsuite';
 
-function createApp(createOptions: HapiPluginOptions) {
+function createApp(createOptions) {
```
Comment (DxCx) [inline packages/graphql-server-hapi/src/hapiApollo.test.ts]: Why removing the typings...?

Comment (alanchristensen) [inline packages/graphql-server-hapi/src/hapiApollo.test.ts]: Hi DxCx,
createOptions can be either a HapiPluginOptions or GraphiQLPluginOptions now. My options were to either create a new type just for this test function that contained a graphqlOptions and graphiqlOptions or remove the type.

Is there another option that I'm missing? My typescript is pretty weak.

Comment (alanchristensen) [inline packages/graphql-server-hapi/src/hapiApollo.test.ts]: Aha, that worked. I had to cast and check for the presence of .graphqlOptions and .graphiqlOptions further below.

**Comment by DxCx** [comment]
@helfer lets merge this as well as soon as @alachristensen is rebasing the code upon latest :)

**Comment by DxCx** [comment]
0.5.1 branch is merged :)
@alanchristensen can you rebase? or i'll?

**Comment by alanchristensen** [comment]
@DxCx I've rebased it.

**Comment by DxCx** [comment]
thanks @alanchristensen :)
@helfer can we merge?

**Comment by helfer** [comment]
Thanks @alanchristensen ! 🙂

**Pull request merged**